lib/phoenix/transports/websocket.ex

```diff
@@ -31,12 +31,19 @@ defmodule Phoenix.Transports.WebSocket do
 
         case handler.connect(config) do
           {:ok, state} -> {:ok, conn, state}
-          :error -> {:error, Plug.Conn.send_resp(conn, 403, "")}
+          :error -> {:error, resp(conn, 403, [], "")}
+          {:error, status, headers, body} -> {:error, resp(conn, status, headers, body)}
```

Comment:
I believe you also need to change the long polling transport.

Reply:
Actually, if we remeber that sockets and channels are meant to be transport agnostic, this approach introduces an issue because we are adding HTTP semantics to the socket. So if you have a non-HTTP transport, the HTTP response won't be useless. Maybe the best option is to move this to transport but I have no idea how to do so at the moment.

Reply:
Oh! Just to clarify, is the problem the `headers` bit? Because as you can see, right now it sends `Plug.Conn.send_resp(conn, 400, "")` which is already using the HTTP status code and body. Just want to understand the problem better :)

Reply:
FYI I've added a test for the long polling transport - didn't have to change any code to make it work.

Reply:
Nevermind, I think I understand the problem now. The `Socket` is supposed to be transport agnostic but by passing the error details through it, that's no longer the case. So if we find a way to move the HTTP-specific error-handling _into_ the Websocket module, we'd fix that.

Reply:
After tinkering with it for a bit I see two options:

(a) adding an option
```
websocket: [
    connection_error_handler: CustomConnectionErrorHandler,
```
to provide your own error handling 

(b) allowing users to provide their own `websocket` transport implementation

```
websocket: [
    implementation: MyWebsocketTransport
```

What do you think

Reply:
Yes, I think something like: `websocket: [error_handler: {module, function, args}]` that is called with the `conn` and must return a `conn` can work. WDYT?

Reply:
I'll make that change 👍